src/engine/strat_engine/setup.rs

```diff
@@ -26,6 +26,67 @@ use super::range_alloc::RangeAllocator;
 use super::serde_structs::PoolSave;
 
 
+/// Determine if devnode is a stratis device, if it is we will add  pool uuid and device
+/// information to pool_map and return pool uuid.
+pub fn is_stratis_device(pool_map: &mut HashMap<PoolUuid, HashMap<Device, PathBuf>>,
+                         devnode: PathBuf,
+                         devno: u64)
+                         -> EngineResult<Option<PoolUuid>> {
+
+    let f = OpenOptions::new().read(true).open(&devnode);
+
+    // There are some reasons for OpenOptions::open() to return an error
+    // which are not reasons for this method to return an error.
+    // Try to distinguish. Non-error conditions are:
+    //
+    // 1. ENXIO: The device does not exist anymore. This means that the device
+    // was volatile for some reason; in that case it can not belong to
+    // Stratis so it is safe to ignore it.
+    //
+    // 2. ENOMEDIUM: The device has no medium. An example of this case is an
+    // empty optical drive.
+    //
+    // Note that it is better to be conservative and return with an
+    // error in any case where failure to read the device could result
+    // in bad data for Stratis. Additional exceptions may be added,
+    // but only with a complete justification.
+    if f.is_err() {
+        let err = f.unwrap_err();
```

Comment:
Please use a separate commit to change ```unwrap_err()``` to ```expect_err("f.is_err()")```. We use expect function when something can be proved to be true. We probably didn't know that ```expect_err()``` existed when this code was written, hence the oversight.